c10/test/core/SymInt_test.cpp

```diff
@@ -0,0 +1,26 @@
+#include <gtest/gtest.h>
+
+#include <c10/core/SymInt.h>
+#include <c10/core/SymbolicIntNode.h>
+
+using namespace c10;
+
+void check(int64_t value) {
+  const auto i = SymInt(value);
+  EXPECT_FALSE(i.is_symbolic());
+  EXPECT_EQ(i.data(), value);
+}
+
+TEST(SymIntTest, ConcreteInts) {
+  check(INT64_MAX);
+  check(0);
+  check(-1);
+  // This is 2^62, which is the most negative number we can support.
+  check(-4611686018427387904);
```

Comment:
nit: can we stick the LL suffix to make sure a compiler won't discard the higher 32bits of this literal?